src/semantics/Action.re

```diff
@@ -2353,12 +2349,12 @@ let rec perform_syn =
       switch (e1) {
       | Tm(_, EmptyHole(_)) =>
         let (rule_p, u_gen) = UHPat.new_EmptyHole(u_gen);
-        let (rule_e, u_gen) = UHExp.new_EmptyHole(u_gen);
+        let (rule_e, ___u_gen) = UHExp.new_EmptyHole(u_gen);
         let rule = UHExp.Rule(rule_p, rule_e);
         ZExp.Deeper(NotInHole, CaseZE(ze1, [rule], Some(Hole)));
       | _ =>
         let (zp, u_gen) = ZPat.new_EmptyHole(u_gen);
-        let (rule_e, u_gen) = UHExp.new_EmptyHole(u_gen);
+        let (rule_e, ___u_gen) = UHExp.new_EmptyHole(u_gen);
```

Comment:
The fact that these u_gens are *supposed* to shadow each other, despite the last instance not being used, is a tricky thing to handle. I'm assuming if this code ever needs to be touched the reader will notice that they need to remove these underscores.

Reply:
leave a comment to clarify what's going on here (maybe @dmoon1221 should do that)

Reply:
I think this [commit](https://github.com/hazelgrove/hazel/commit/92c775eabeb08d832b2946a301299b177be1948e) resolves this.